Jobs get stuck when a Job Listener throws an exception #918

plotnikovma opened the pull request:
<!--
If this is your first time contributing to the project (or it's been a while), please consider reviewing https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md
-->


This PR...

Fixes issue #

## Changes
-

-----------------
## Checklist
- [ ] tested locally
- [ ] updated the docs
- [ ] added appropriate test
- [ ] signed-off on the DCO referenced in the CONTRIBUTING link below via `git commit -s` on my commits, and submit this code under terms of the Apache 2.0 license and assign copyright to the Quartz project owners
  (If you're not using command-line, you can use a [browser extension](https://github.com/scottrigby/dco-gh-ui) )
-----------------
In submitting this contribution, I agree to the terms of contributing as referred to here: 
https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md

jhouserizer commented:
Hi!  Thanks for contributing.
Can you please add a little context around this?  I see in the changes that you are sending the JobExecutionContect to the various listeners by putting a handle to it on the SchedulerException ...  But that's all I see.  I don't see how this works around anything (at least not directly).

plotnikovma commented:
```
Hi! Thanks for contributing.
Can you please add a little context around this? I see in the changes that you are sending the JobExecutionContect to the various listeners by putting a handle to it on the SchedulerException ... But that's all I see. I don't see how this works around anything (at least not directly).
```
Hi, I will update the information soon

plotnikovma commented:
@jhouserizer 
hi, I added new commits and description.

After studying the code, I realized that errors should be corrected in the SchedulerListener#schedulerError() method, but this method only receives a text error that can only be logged. I added context transfer so that when an error is intercepted, you can understand what trigger or task it occurred with and try to fix the task freezing.
